### PR TITLE
Fixing account id references for public app local dev

### PR DESCRIPTION
**Pull request by brandenrodgers**
## Description and Context

This fixes a few different small edge-case bugs, all related to local dev for public apps:
- We were showing the incorrect account id in the prompt when the user edits a configuration file while local dev is running
- We were showing the incorrect account to upload to when there is no deployed build for the project in the target account
  - This only shows when the project exists in the account, but there are no builds (repro by uploading a broken project)
- Fixing JS error when default account is a dev test account, and the parent dev account has not been configured in the CLI

## Screenshots

WIP copy for the message to tell the user that they need to configure their Developer Account:
image

## Who to Notify

cc/ @markhazlewood for input on copy

### CODE REVIEW

File: packages/cli/commands/project/dev.js [hunk 1]
```diff
@@ -104,12 +108,30 @@ exports.handler = async options => {
...
+              `hs auth --account=${accountConfig.parentAccountId}`
+            ),
+          })
+        );
+        process.exit(EXIT_CODES.ERROR);
+      }
+      targetProjectAccountId = accountConfig.parentAccountId;
+    } else {
+      targetProjectAccountId = accountId;
+    }
```
Comment (camden11) [inline packages/cli/commands/project/dev.js:134]: I think this could be pulled into `lib/localDev`

**Review by camden11 — APPROVED** [review]
LGTM, one minor suggestion

File: packages/cli/commands/project/dev.js (hunk 1 above)
Comment (brandenrodgers) [reply packages/cli/commands/project/dev.js:134]: @camden11 just to clarify, which part do you think should be pulled in? All this logic is really doing is checking to see if the parent account exists in the config or not.

Comment (brandenrodgers) [reply packages/cli/commands/project/dev.js:134]: I spent a little time trying to do this, but I couldn't get something I was happy with. There are so many args that need to be passed into the abstracted util that it ends up not feeling very valuable. I'm going to merge this as-is, but feel free to tweak this later on if you can think of a cleaner format for it!

File: packages/cli/lang/en.lyaml [hunk 2]
```diff
@@ -479,6 +479,7 @@ en:
             errors:
               noProjectConfig: "No project detected. Please run this command again from a project directory."
               invalidProjectComponents: "Projects cannot contain both private and public apps. Move your apps to separate projects before attempting local development."
+              parentAccountNotConfigured: "To develop this project locally, run {{ authCommand }} to authenticate the App Developer Account {{ accountId}} associated with {{ accountIdentifier }}."
```
Comment (brandenrodgers) [inline packages/cli/lang/en.lyaml:482]: cc/ @adamawang for 👍 on the copy here

File: packages/cli/lang/en.lyaml [hunk 3]
```diff
@@ -479,6 +479,7 @@ en:
             errors:
               noProjectConfig: "No project detected. Please run this command again from a project directory."
               invalidProjectComponents: "Projects cannot contain both private and public apps. Move your apps to separate projects before attempting local development."
+              parentAccountNotConfigured: "To develop this project locally, run {{ authCommand }} to authenticate the App Developer Account {{ accountId }} associated with {{ accountIdentifier }}."
```
Comment (adamawang) [inline packages/cli/lang/en.lyaml:482]: 💎 works for us!

**Pull request merged**